test/env/lib.rs

```diff
@@ -147,8 +148,35 @@ impl Transfer {
     }
 }
 
+#[async_trait(?Send)]
+pub trait LedgerEnv {
+    fn fork(&self) -> Self;
+    fn principal(&self) -> Principal;
+    async fn query<Input, Output>(&self, method: &str, input: Input) -> anyhow::Result<Output>
+    where
+        Input: ArgumentEncoder + std::fmt::Debug,
+        Output: for<'a> ArgumentDecoder<'a>;
+    async fn update<Input, Output>(&self, method: &str, input: Input) -> anyhow::Result<Output>
+    where
+        Input: ArgumentEncoder + std::fmt::Debug,
+        Output: for<'a> ArgumentDecoder<'a>;
+}
+
+#[async_trait(?Send)]
+pub trait LedgerEndpoint: LedgerEnv {
```

Comment:
I'm not sure this should be a trait. The logic of all ICRC-1 and ICRC-2 methods should be the same except for the `LedgerEnv`. I would expect functions like `transfer` to not be part of a trait and instead take in input the `LedgerEnv`.

Reply:
That also works for me; I don't have strong preferences.